File: SZE/OODB/20181017/Plotter/Circle.cpp
#include "Circle.hpp"

Circle::Circle(double x, double y, double r, char s)
  : Shape(s), center(x,y), radius(r) {}

Circle::Circle(Point c, double r, char s)
  : Shape(s), center(c), radius(r) {}

Circle::~Circle(){}

bool Circle::contains(const Point& p) const {
  return p-center <= radius;
}

File: FHWN/OO/2026_09_16_Practice_session/SVG_creator/Circle.py
from Point import *
from Shape import *

class Circle(Shape):
    __center: Point
    __radius: float
    __color: str

    def __init__(self, center:Point, radius:float, color:str = "black"):
        self.__center = center
        self.__radius = radius
        self.__color = color
        
    def area(self) -> float:
        return self.__radius ** 2 * 3.141521

    def to_svg(self) -> float:
        return f'<circle cx="{self.__center.x}" cy="{self.__center.y}" r="{self.__radius}" fill="{self.__color}" />'


